tests/test_dirhash.py

```diff
@@ -282,12 +282,45 @@ def test_ignore_hidden_files(self):
         assert filepaths == ['.d2/f1', '.f2', 'd1/.f2', 'd1/f1', 'f1']
 
         # with ignore
-        filepaths = included_paths(
-            self.path_to('root'),
-            match=['*', '!.*']
-        )
+        filepaths = included_paths(self.path_to('root'), match=['*', '!.*'])
+        assert filepaths == ['d1/f1', 'f1']
+
+    def test_ignore_hidden_files_only(self):
```

Comment:
2. test - after reading through suggested glob documentation, i was able to figure out combination of patterns that allow the original ignore result including visible file inside of hidden folder '.d2/f1'